src/tracing/writer.rs

```diff
@@ -319,9 +325,9 @@ impl<W: Write> TraceWriter<W> {
         let log_style = self.log_style();
         self.write_branch()?;
 
-        if let Some(name) = &log.decoded.name {
+        if let Some(name) = &log.decoded.as_ref().unwrap().name {
             write!(self.writer, "emit {name}({log_style}")?;
-            if let Some(params) = &log.decoded.params {
+            if let Some(params) = &log.decoded.as_ref().unwrap().params {
```

Comment:
here we can also use a new helper fn that returns an option, then we can avoid the unwrap